### ISSUE TITLE
Unable to add account 

### ISSUE THREAD

**Issue by BhavanaPilli11** [microsoft/azuredatastudio#20262]
Issue Type: <b>Bug</b>

Unable to add account and it fails with error  "no access token returned from microsoft oauth"
username - b0p00b9@homeoffice.wal-mart.com

Azure Data Studio version: azuredatastudio 1.38.0 (dff93c4736269ef2a7e8a225737a0d8354bf0246, 2022-07-26T17:24:38.381Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2800)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 3|
|Memory (System)|16.00GB (0.08GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

**Comment by BhavanaPilli11** [comment]
[1-Azure Accounts.log](https://github.com/microsoft/azuredatastudio/files/9339798/1-Azure.Accounts.log)

**Comment by BhavanaPilli11** [comment]
Followed above steps and attached the logs file above. Please check and let me know if any other info is required.

**Comment by BhavanaPilli11** [comment]
No access token returned from Microsoft OAuth
onDidChangeNotification @ workbench.desktop.main.js:1635
(anonymous) @ workbench.desktop.main.js:1635
fire @ workbench.desktop.main.js:137
addNotification @ workbench.desktop.main.js:1637
notify @ workbench.desktop.main.js:2419
(anonymous) @ workbench.desktop.main.js:2369
ZoneAwarePromise @ /Applications/Azure Data Studio.app/Contents/Resources/app/node_modules.asar/zone.js/dist/zone.js:1340
_showMessage @ workbench.desktop.main.js:2369
$showMessage @ workbench.desktop.main.js:2369
_doInvokeHandler @ workbench.desktop.main.js:2378
_invokeHandler @ workbench.desktop.main.js:2378
_receiveRequest @ workbench.desktop.main.js:2378
_receiveOneMessage @ workbench.desktop.main.js:2378
(anonymous) @ workbench.desktop.main.js:2378
fire @ workbench.desktop.main.js:137
fire @ workbench.desktop.main.js:184
_receiveMessage @ workbench.desktop.main.js:184
(anonymous) @ workbench.desktop.main.js:184
fire @ workbench.desktop.main.js:137
acceptChunk @ workbench.desktop.main.js:184
(anonymous) @ workbench.desktop.main.js:184
k @ workbench.desktop.main.js:4048
emit @ events.js:315
addChunk @ internal/streams/readable.js:309
readableAddChunk @ internal/streams/readable.js:284
Readable.push @ internal/streams/readable.js:223
onStreamRead @ internal/stream_base_commons.js:188
workbench.desktop.main.js:854   ERR Cannot read property 'accountType' of undefined: TypeError: Cannot read property 'accountType' of undefined
    at rl.renderElement (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:856:39117)
    at rl.renderElement (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:856:39771)
    at K.renderElement (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:264:1778)
    at y.insertItemInDOM (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:246:18608)
    at y._splice (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:246:14505)
    at y.splice (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:246:12767)
    at vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:67:67112
    at Array.forEach (<anonymous>)
    at t.splice (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:67:67099)
    at vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:264:8541
    at d.bufferEvents (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:137:4757)
    at ie.splice (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:264:8508)
    at W.updateAccounts (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3739:83009)
    at z.updateProviderAccounts (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3739:88846)
    at vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3739:84304
    at n.fire (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:137:1712)
    at vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:854:15122
    at n.fire (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:137:1712)
    at Kd.fireAccountListUpdate (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3739:97423)
    at vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3739:93540
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async yo.run (vscode-file://vscode-app/Applications/Azure%20Data%20Studio.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:856:13900)

**Comment by BhavanaPilli11** [comment]
Above are the error logs in Developer tool console.

**Comment by kburtram** [comment]
thanks @BhavanaPilli11!  @cssuh can you please take a look at the above error and provide a recommendation for possible workarounds?

**Comment by BhavanaPilli11** [comment]
@cssuh @kburtram any update on the issue?

**Comment by cssuh** [comment]
Hi @BhavanaPilli11 have you been able to authenticate successfully in the past? Can you please try removing your account and re-adding it?

**Comment by BhavanaPilli11** [comment]
@cssuh I was never able to authenticate successfully. I don't have any accounts added.

**Comment by cssuh** [comment]
@BhavanaPilli11 Can you try using Device Code Auth to get you unblocked for now? We'll continue looking into this issue.

**Comment by cssuh** [comment]
@BhavanaPilli11 @aholler2 this is probably due to some proxy configuration, have you set the proxy configuration in ADS?

**Comment by aholler2** [comment]
@cssuh http_proxy and https_proxy variables are both configured

**Comment by cssuh** [comment]
Hi @aholler2 , are your proxy variables configured as follows:
```
export http_proxy="http://userName@fqdn:yourPassword@yourProxyURL.com:8080"
export https_proxy="http://userName@fqdn:yourPassword@yourProxyURL.com:8080"
```

**Comment by aholler2** [comment]
@cssuh Yes, they are both configured. They don't need authentication with username/password.

PS C:\Users\hollea> dir env:

Name                           Value
----                           -----
http_proxy                     http://proxy.mycompanydomain.com:80
https_proxy                    http://proxy.mycompanydomain.com:80

**Comment by cssuh** [comment]
@aholler2 
You also need to whitelist the following URLs:

https://management.azure.com/
https://login.microsoftonline.com/

you may need to whitelist additional URLs but this can sometimes vary on a case-by-case basis - in order to verify you aren’t blocking any URLs from going through, go to Help > Toggle Developer Tools and select the Network tab. Here you will see any URLs that are getting blocked that you may need to whitelist to successfully add your account.

**Comment by cssuh** [comment]
Not able to repro, closing as there has been no activity. Please feel free to open a new issue or re-open this one if you are still seeing problems
